Sources/NIOSSL/NIOSSLHandler.swift

```diff
@@ -102,14 +107,18 @@ public class NIOSSLHandler : ChannelInboundHandler, ChannelOutboundHandler, Remo
     }
 
     public func channelActive(context: ChannelHandlerContext) {
+        context.channel.getOption(ChannelOptions.allowRemoteHalfClosure).whenSuccess { halfClosureAllowed in
```

Comment:
I don't think caching this value is semantically quite right: users can change this value during the lifetime of the `Channel`. We should look it up when we need it.

Similarly, we should try to use the sync options where we can.